src/font.c

```diff
@@ -554,27 +554,39 @@ static void gl_printHalo( const glFont *ft_font,
          const glColour*, const char*, const int ))
 {
    const glColour *bg;
+   double radius;
+   double halfRadius;
+
    /* TODO: This method works, but is inefficient. Should probably be
-    * ultimately replaced with use of signed distance fields or some
-    * other more efficient method (signed distance fields seem to be
-    * the "right" solution). */
+   * ultimately replaced with use of signed distance fields or some
+   * other more efficient method (signed distance fields seem to be
+   * the "right" solution). */
    if ((c == NULL) || (c->a >= 1.)) {
-      if ((c==NULL) || (c->r+c->b+c->g > 1.5))
-         bg = &cBlack;
-      else
-         bg = &cWhite;
-      func( ft_font, width, height, bx - 1, by, bg, text, 1 );
-      func( ft_font, width, height, bx + 1, by, bg, text, 1 );
-      func( ft_font, width, height, bx, by - 1, bg, text, 1 );
-      func( ft_font, width, height, bx, by + 1, bg, text, 1 );
-      func( ft_font, width, height, bx - 2, by, bg, text, 1 );
-      func( ft_font, width, height, bx + 2, by, bg, text, 1 );
-      func( ft_font, width, height, bx, by - 2, bg, text, 1 );
-      func( ft_font, width, height, bx, by + 2, bg, text, 1 );
-      func( ft_font, width, height, bx - 1, by - 1, bg, text, 1 );
-      func( ft_font, width, height, bx - 1, by + 1, bg, text, 1 );
-      func( ft_font, width, height, bx + 1, by - 1, bg, text, 1 );
-      func( ft_font, width, height, bx + 1, by + 1, bg, text, 1 );
+      if ((c==NULL) || (c->r+c->b+c->g > 2.8)) {
+        bg = &cBlack;
+        radius = 1.2;
+      } else if ((c->r+c->b+c->g > 1.5)){
+        bg = &cGrey10;
+        radius = 1.2;
+      } else {
+        bg = &cWhite;
+        radius = 1.2;
+      }
+
+      halfRadius = radius / 2;      
+
+      func( ft_font, width, height, bx - halfRadius, by, bg, text, 1 );
+      func( ft_font, width, height, bx + halfRadius, by, bg, text, 1 );
+      func( ft_font, width, height, bx, by - halfRadius, bg, text, 1 );
+      func( ft_font, width, height, bx, by + halfRadius, bg, text, 1 );
+      func( ft_font, width, height, bx - radius, by, bg, text, 1 );
+      func( ft_font, width, height, bx + radius, by, bg, text, 1 );
+      func( ft_font, width, height, bx, by - radius, bg, text, 1 );
+      func( ft_font, width, height, bx, by + radius, bg, text, 1 );
+      func( ft_font, width, height, bx - halfRadius, by - halfRadius, bg, text, 1 );
+      func( ft_font, width, height, bx - halfRadius, by + halfRadius, bg, text, 1 );
+      func( ft_font, width, height, bx + halfRadius, by - halfRadius, bg, text, 1 );
+      func( ft_font, width, height, bx + halfRadius, by + halfRadius, bg, text, 1 );
```

Comment:
Ah, it looks like indentation here was messed up again with your recent changes (the inner stuff is indented by 2 instead of 1 and the comment has the little asterisks misaligned)

🕷️